client/views/view_cities.cpp

```diff
@@ -732,6 +705,47 @@ void city_widget::display_list_menu(const QPoint)
   list_menu->popup(QCursor::pos());
 }
 
+/**
+   Sell building for all selected cities after asking for
+   confirmation.
+*/
+void city_widget::sell(const struct impr_type *building)
+{
+  fc_assert_ret(building != nullptr);
+
+  Impr_type_id impr_id = improvement_number(building);
+  if (nullptr == improvement_by_number(impr_id)) {
+    return;
+  }
+
+  hud_message_box *ask = new hud_message_box(king()->central_wdg);
+  QString imprname = improvement_name_translation(building);
+  QString buf =
+      QString(_("Are you sure you want to sell the %1?")).arg(imprname);
+  ask->setStandardButtons(QMessageBox::Cancel | QMessageBox::Yes);
+  ask->setDefaultButton(QMessageBox::No);
+  ask->button(QMessageBox::Yes)->setText(_("Yes Sell"));
+  ask->set_text_title(buf, _("Sell?"));
+  ask->setAttribute(Qt::WA_DeleteOnClose);
+  connect(ask, &hud_message_box::accepted, this, [=]() {
+    const auto saved_selection =
+        selected_cities; // Copy to avoid invalidations
+
+    for (auto *pcity : saved_selection) {
+      int city_id = pcity->id;
+      if (nullptr == game_city_by_number(city_id)) {
+        return;
```

Comment:
I think skipping the lost city makes more sense in this case.
```suggestion
        continue;
```